wrapping/itkBlockMatchingMultiResolutionImageRegistrationMethod.wrap

```diff
@@ -0,0 +1,61 @@
+set(WRAPPER_AUTO_INCLUDE_HEADERS OFF)
+itk_wrap_include("itkPoint.h")
+itk_wrap_include("itkSmartPointer.h")
+itk_wrap_include("itkImage.h")
+itk_wrap_include("itkImageRegion.h")
+
+# Wrap base classes for itk::BlockMatching::MetricImageToDisplacementCalculator public members
+itk_wrap_class("itk::Image" POINTER)
+  foreach(t ${WRAP_ITK_REAL})
+    foreach(d ${ITK_WRAP_IMAGE_DIMS})
+      itk_wrap_template("P${ITKM_${t}}${d}${d}" "itk::Point<${ITKT_${t}},${d}>, ${d}")
+      itk_wrap_template("SPI${ITKM_${t}}${d}${d}" "itk::SmartPointer<itk::Image<${ITKT_${t}},${d}>>, ${d}")
+    endforeach()
+  endforeach()
+itk_end_wrap_class()
```

Comment:
These `Image` wrappings are required for `itkBlockMatchingMetricImageToDisplacementCalculator` public members of type [`'MetricImageImageType`](https://github.com/KitwareMedical/ITKUltrasound/blob/master/include/itkBlockMatchingMetricImageToDisplacementCalculator.h#L91). However, even after adding this wrapping we're seeing warnings about unknown parameters:

> Warning: Unknown parameter 'itk::SmartPointer<itk::Image<float,2>>' in template 'itk::Image'
Warning: Unknown parameter 'itk::SmartPointer<itk::Image<float,3>>' in template 'itk::Image'
Warning: Unknown parameter 'itk::SmartPointer<itk::Image<double,2>>' in template 'itk::Image'
Warning: template already defined 'itk::Image<itk::SmartPointer<itk::Image<double,2>>,2>'
Warning: Unknown parameter 'itk::SmartPointer<itk::Image<double,3>>' in template 'itk::Image'
Warning: template already defined 'itk::Image<itk::SmartPointer<itk::Image<double,3>>,3>'

This may be an issue with `SmartPointer` not being wrapped, will revisit shortly.

Reply:
Attempting to wrap `SmartPointer` yields compile time errors. Could this be an issue with circular dependencies?

> 16>C:\repos\itkus\Wrapping\Modules\Ultrasound\itkBlockMatchingMultiResolutionImageRegistrationMethodPython.cpp(42678,30): error C3861: 'itkImageD2_cast': identifier not found
16>C:\repos\itkus\Wrapping\Modules\Ultrasound\itkBlockMatchingMultiResolutionImageRegistrationMethodPython.cpp(43971,16): error C3861: 'itkImageBase2_TransformPhysicalPointToIndex': identifier not found
16>C:\repos\itkus\Wrapping\Modules\Ultrasound\itkBlockMatchingMultiResolutionImageRegistrationMethodPython.cpp(43995,3): error C2065: 'itkContinuousIndexD2': undeclared identifier
16>C:\repos\itkus\Wrapping\Modules\Ultrasound\itkBlockMatchingMultiResolutionImageRegistrationMethodPython.cpp(43995,24): error C2146: syntax error: missing ';' before identifier 'result'
16>C:\repos\itkus\Wrapping\Modules\Ultrasound\itkBlockMatchingMultiResolutionImageRegistrationMethodPython.cpp(43995,24): error C2065: 'result': undeclared identifier
16>C:\repos\itkus\Wrapping\Modules\Ultrasound\itkBlockMatchingMultiResolutionImageRegistrationMethodPython.cpp(44037,7): error C2065: 'result': undeclared identifier
16>C:\repos\itkus\Wrapping\Modules\Ultrasound\itkBlockMatchingMultiResolutionImageRegistrationMethodPython.cpp(44037,16): error C3861: 'itkImageBase2_TransformPhysicalPointToContinuousIndex': identifier not found
16>C:\repos\itkus\Wrapping\Modules\Ultrasound\itkBlockMatchingMultiResolutionImageRegistrationMethodPython.cpp(44046,15): error C2061: syntax error: identifier 'itkContinuousIndexD2'
...